source/sessions/tests/README.rst

```diff
@@ -178,7 +178,7 @@ Skip this test if your driver does not allow forking.
 14. Implicit sessions only allocate their server session after a successful connection checkout
 ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
 
-* Create a MongoClient with the following options: ``maxPoolSize=1`` and ``retryWrites=true``
+* Create a MongoClient with the following options: ``maxPoolSize=1`` and ``retryWrites=true``. If testing against a sharded deployment, be sure to connect to only a single mongos.
```

Comment:
Is this possibly related to the need to run the test 5 times to be able to assert that only 1 session is used? If we're limiting the number of `mongos` nodes to 1, are there still cases where more than 1 session is used to run the concurrent operations in the test?

Reply:
IIUC it is still possible to have more than one session checked out because part of the requirement here is to ensure that we end the implicit sessions _after_ we close the connection. The order of operations looks like this:

1. make connection
2. check out implicit session
3. close connection
4. end implicit session

For concurrent jobs I believe there is always a race condition between steps 2 and 4. In the Go driver, we can dependably illustrate this by creating some kind of delay like `time.Sleep(1)` in the `(session.Client).EndSession` method.

Reply:
That's right, we still need to ru-run the test even with one mongos.

Reply:
Sounds good, thanks for the explanation 👍